ClientSimulator/src/main/java/client/Client.java

```diff
@@ -153,7 +153,15 @@ public void close() {
         System.out.println("Logged out.");
     }
 
-    public void waitForMarketDataUpdate() { while(!mktDataUpdateSemaphore.acquire()){} }
+    public void waitForMarketDataUpdate() {
+        long startTime = System.currentTimeMillis();
+        while(!mktDataUpdateSemaphore.acquire()){
```

Comment:
acquire is a blocking operation. This should be tryAcquire with a timeout

Reply:
No problem. This was the implementation in the Hawkes client so does that need to be changed to use a tryAquire as well?